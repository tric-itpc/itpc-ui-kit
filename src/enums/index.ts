export enum UIKitColors {
  black = "#000",
  green = "#4DB04D",
  grey = "#B2B7C7",
  greyLight = "#E5E5E5",
  purple = "#5C53AC",
  purpleLight = "#6F65CB",
  red = "#D42564",
  redDark = "#C9184A",
  white = "#FFF",
  yellow = "#DCB21E",
}

export enum Theme {
  DARK = "dark",
  DEFAULT = "default",
}

export enum SortType {
  ASCENDING = "ascending",
  DESCENDING = "descending",
  NONE = "none",
}

export enum NumberColumns {
  ONE = "one",
  TWO = "two",
  ZERO = "zero",
}

export enum RowAlign {
  BOTTOM = "bottom",
  MIDDLE = "middle",
  TOP = "top",
}

export enum RowJustify {
  CENTER = "center",
  END = "end",
  SPACE_AROUND = "around",
  SPACE_BETWEEN = "between",
  SPACE_EVENLY = "evenly",
  START = "start",
}

export enum TextSize {
  L = "l",
  M = "m",
  S = "s",
  XL = "xl",
  XS = "xs",
  XXL = "xxl",
  XXS = "xxs",
}

export enum TextWeight {
  BOLD = "bold",
  LIGHT = "light",
  MEDIUM = "medium",
  NORMAL = "normal",
}

export enum TextType {
  DANGER = "danger",
  DISABLED = "disabled",
  PRIMARY = "primary",
  SECONDARY = "secondary",
  SUCCESS = "success",
  WARNING = "warning",
}

export enum TextTag {
  CODE = "code",
  EM = "em",
  PARAGRAPH = "p",
  STRONG = "strong",
}

export enum AnimationTypes {
  FIXED = "FIXED",
  TRANSFORM_BY_AXIS_X = "transform-by-axis-x",
  TRANSFORM_BY_AXIS_X_Y = "transform-by-axis-x_y",
}
