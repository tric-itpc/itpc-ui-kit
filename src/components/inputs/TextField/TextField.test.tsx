import React from "react"

import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"

import { AutoComplete, KeyCode } from "../../../enums"
import {
  blurInput,
  changeInput,
  focusInput,
  getInputAttributes,
  getInputAttributeValue,
  getInputDisabled,
  getInputError,
  getInputValue,
  keyDownInput,
} from "../../../lab/tests"

import { TextField } from "./index"

describe("Компонент TextField", () => {
  const defaultProps = {
    id: "test-input",
    name: "test-input",
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onFocus: jest.fn(),
    onKeyDown: jest.fn(),
    placeholder: "Введите текст",
  }

  test("Рендерится с дефолтными пропсами (снимок)", () => {
    const { container } = render(
      <TextField {...defaultProps} data-testid="text-field" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("Отображает placeholder", () => {
    render(<TextField {...defaultProps} data-testid="text-field" />)
    const placeholder = screen.getByText("Введите текст")
    expect(placeholder).toBeInTheDocument()
  })

  test("Отображает переданное значение", () => {
    render(
      <TextField
        {...defaultProps}
        data-testid="text-field"
        value="Тестовый текст"
      />
    )
    const input = screen.getByTestId("text-field")
    expect(getInputValue(input)).toBe("Тестовый текст")
  })

  test("Вызывает onChange при изменении значения", () => {
    const onChange = jest.fn()
    render(
      <TextField
        {...defaultProps}
        data-testid="text-field"
        onChange={onChange}
      />
    )
    const input = screen.getByTestId("text-field")
    changeInput(input, "Новый текст")

    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toHaveBeenCalledWith("Новый текст", expect.any(Object))
  })

  test("Вызывает onFocus при фокусе", () => {
    const onFocus = jest.fn()
    render(
      <TextField {...defaultProps} data-testid="text-field" onFocus={onFocus} />
    )
    const input = screen.getByTestId("text-field")
    focusInput(input)

    expect(onFocus).toHaveBeenCalledTimes(1)
  })

  test("Вызывает onBlur при потере фокуса", () => {
    const onBlur = jest.fn()
    render(
      <TextField {...defaultProps} data-testid="text-field" onBlur={onBlur} />
    )
    const input = screen.getByTestId("text-field")
    focusInput(input)
    blurInput(input)

    expect(onBlur).toHaveBeenCalledTimes(1)
  })

  test("Отображает ошибку, если validationState = 'invalid'", () => {
    const errorMessage = "Ошибка ввода"
    render(
      <TextField
        {...defaultProps}
        data-testid="text-field"
        errorMessage={errorMessage}
        validationState="invalid"
      />
    )
    const error = screen.getByText(errorMessage)
    expect(error).toBeInTheDocument()
  })

  test("Не отображает ошибку, если validationState = 'valid'", () => {
    const errorMessage = "Ошибка ввода"
    render(
      <TextField
        {...defaultProps}
        data-testid="text-field"
        errorMessage={errorMessage}
        validationState="valid"
      />
    )
    const input = screen.getByTestId("text-field")
    const error = getInputError(input)
    expect(error).not.toHaveClass("itpc-error_show")
  })

  test("Применяет класс disabled, если disabled = true", () => {
    render(<TextField {...defaultProps} data-testid="text-field" disabled />)
    const input = screen.getByTestId("text-field")
    expect(getInputDisabled(input)).toBeTruthy()
  })

  test("Обрабатывает нажатие клавиш", () => {
    render(<TextField {...defaultProps} data-testid="text-field" />)
    const input = screen.getByTestId("text-field")
    keyDownInput(input, KeyCode.ENTER)

    expect(defaultProps.onKeyDown).toHaveBeenCalledTimes(1)
  })

  test("Отображает иконку, если она передана", () => {
    const icon = <span>🔍</span>
    render(<TextField {...defaultProps} data-testid="text-field" icon={icon} />)
    const iconElement = screen.getByText("🔍")
    expect(iconElement).toBeInTheDocument()
  })

  test("Применяет maxLength, если он передан", () => {
    render(
      <TextField {...defaultProps} data-testid="text-field" maxLength={10} />
    )
    const input = screen.getByTestId("text-field")
    expect(getInputAttributes(input)).toHaveProperty("maxlength")
    expect(getInputAttributeValue(input, "maxlength")).toEqual("10")
  })

  test("Применяет autoComplete, если он передан", () => {
    render(
      <TextField
        {...defaultProps}
        autoComplete={AutoComplete.OFF}
        data-testid="text-field"
      />
    )
    const input = screen.getByTestId("text-field")
    expect(getInputAttributes(input)).toHaveProperty("autocomplete")
    expect(getInputAttributeValue(input, "autocomplete")).toEqual(
      AutoComplete.OFF
    )
  })

  test("Применяет кастомный класс, если он передан", () => {
    render(
      <TextField
        {...defaultProps}
        className="custom-class"
        data-testid="text-field"
      />
    )
    const input = screen.getByTestId("text-field")
    expect(input).toHaveClass("custom-class")
  })
})
