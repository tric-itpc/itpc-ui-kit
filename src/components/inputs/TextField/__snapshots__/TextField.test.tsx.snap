// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент TextField Рендерится с дефолтными пропсами (снимок) 1`] = `
<div
  class="itpc-field"
  data-testid="text-field"
>
  <div
    class="itpc-input-wrap itpc-input-wrap_hover"
    style="height: 40px; max-height: 200px;"
  >
    <label
      class="itpc-placeholder"
      for="test-input"
    >
      Введите текст
    </label>
    <input
      class="itpc-input"
      id="test-input"
      name="test-input"
      type="text"
      value=""
    />
    <span
      class="itpc-error"
      title=""
    />
  </div>
</div>
`;
